content/blog/2020/05/2020-05-07-docker-windows-agents.adoc

```diff
@@ -0,0 +1,121 @@
+
+Command line arguments include:
+
+* `-Url <URL>` or JENKINS_URL - the Jenkins master URL (this should NOT include the computer/NAME/slave-agent.jnlp part)
+* `-Secret <SECRET>` or JENKINS_SECRET - the secret as shown on the master after creating the agent
+* `-Name <NAME>` or JENKINS_AGENT_NAME - the name of the agent, it should match the name you specified when creating the agent on the master
+* `-Tunnel <HOST:PORT>` or JENKINS_TUNNEL - A tunnel host to route the TCP traffic through when connecting to the master
+* `-WorkDir <WORKDIR>` or JENKINS_AGENT_WORKDIR - same as the -workDir parameter mentioned above in the jenkins/agent image information.
+* `-WebSocket` or JENKINS_WEB_SOCKET=true - when present the connection to the master will be done via WebSocket through the Jenkins URL rather than using a separate network port.
+* `-DirectConnection <HOST:PORT>` or JENKINS_DIRECT_CONNECTION - Connect directly to this TCP agent port, skipping the HTTP(S) connection parameter download.
+* `-InstanceIdentity <BASE64VALUE>` or JENKINS_INSTANCE_IDENTITY - The base64 encoded InstanceIdentity byte array of the Jenkins master. When this is set, the agent skips connecting to an HTTP(S) port for connection info
+* `-JavaHome <JAVA_HOME>` or JAVA_HOME  - An override for the default JAVA_HOME baked into the image (different for JDK8 vs. JDK11)
+* `-Protocols <PROTOCOLS>` or JENKINS_PROTOCOLS - Specify the link:https://github.com/jenkinsci/remoting/blob/de7818885a5bf478760ba29f5ee216291437cb16/docs/protocols.md#active-protocols[remoting protocols] to attempt when instanceIdentity is provided.
+
+[NOTE]
+====
+There is currently an issue where the `-Protocols` parameter is not actually passed to agent.jar. The fix has been merged, but a release has not been made.
+====
+
+Example:
+
+```
+docker run jenkins/inbound-agent:windowsservercore-1809 -Url http://jenkins-server:port -WorkDir=C:/Users/jenkins/Agent -Secret <SECRET> -Name <AGENTNAME>
+```
+
+Example using environment variables:
+
+```
+docker run -e "JENKINS_URL=http://jenkins-server:port" -e "JENKINS_AGENT_NAME=AGENTNAME" jenkins/inbound-agent:windowsservercore-1809 -WorkDir=C:/Users/jenkins/Agent -Secret <SECRET> -Name <AGENTNAME>
+```
+
+The URL, Name and Secret parameters are required, but can be specified as either command line parameters or environment variables.
+
+
+
+=== jenkins/ssh-agent
+
+As mentioned above the jenkins/ssh-agent docker image is based on SSH communication with the master, rather than the remoting TCP or WebSocket protocols. This communication is initiated by the master rather than the agent connecting to the master is done with jenkins/agent and jenkins/inbound-agent.
```

Comment:
Again the most important use case (IMO) for `jenkins/agent` is a DiY outbound agent.

Reply:
Let me know if my updates are ok with regards to your ideas.